clippy_lints/src/manual_div_ceil.rs

```diff
@@ -0,0 +1,127 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet_with_applicability;
+use clippy_utils::SpanlessEq;
+use rustc_ast::{BinOpKind, LitKind};
+use rustc_data_structures::packed::Pu128;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::{self};
+use rustc_session::declare_lint_pass;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for an expression like `(x + (y - 1)) / y` which is a common manual reimplementation
+    /// of `x.div_ceil(y)`.
+    ///
+    /// ### Why is this bad?
+    /// It's simpler and more readable.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// let x: i32 = 7;
+    /// let y: i32 = 4;
+    /// let div = (x + (y - 1)) / y;
+    /// ```
+    /// Use instead:
+    /// ```no_run
+    /// #![feature(int_roundings)]
+    /// let x: i32 = 7;
+    /// let y: i32 = 4;
+    /// let div = x.div_ceil(y);
+    /// ```
+    #[clippy::version = "1.81.0"]
+    pub MANUAL_DIV_CEIL,
+    complexity,
```

Comment:
Some drive-by thoughts 😄 :

It seems like `div_ceil` for signed integers are still unstable, so if this was put in `complexity` group, we might ended up suggesting something that the users cannot fix.
As for unsigned integers, this function was stabilized for 1.73.0 and onward, so a [`msrv`](https://doc.rust-lang.org/clippy/development/adding_lints.html#specifying-the-lints-minimum-supported-rust-version-msrv) configuration would be very nice~

Reply:
Thanks for your input!

I'm struggling to find a proper way to separate lint for stable and unstable (i.e. unsigned and signed integers)
Can you please point me in a direction that would be helpful for this?

Reply:
ah yes, in [`check_literal`](https://github.com/rust-lang/rust-clippy/pull/12987/files#diff-9cf499a7658cd325dadb12d2c2a1bbbdf8742c80909a0132790a10564444b91fR111), 
you can use the type check result on that literal, so it's something like:

```rust
let ty = cx.typeck_results().expr_ty(expr);
```
Which gives you an `rustc_middle::ty::Ty`, and you should be able to extract the info of whether it's an `int` or `uint`

Reply:
BTW, when a binary operand is another binary expression, the suggestion will be broken, something like:

```rust
let z: u32 = 7;
let _ = (z as i32 + (y - 1)) / y;
```

this suggests `z as i32.div_ceil(y)`, you'll need a pair of parentheses to wrap the caller (see [`maybe_par`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/sugg/enum.Sugg.html#method.maybe_par)).

Reply:
> I'm struggling to find a proper way to separate lint for stable and unstable (i.e. unsigned and signed integers) Can you please point me in a direction that would be helpful for this?

ah yes, it appears that you already have a function to [check if it's `Int` or `Uint`](https://github.com/rust-lang/rust-clippy/pull/12987/files#diff-9cf499a7658cd325dadb12d2c2a1bbbdf8742c80909a0132790a10564444b91fR106-R109) right?

Then you should be able to separated two scenarios, one is where the type is `ty::Int`, in this case, don't offer a fixable suggestion yet... (I would say suggest only if the user already have `int_roundings` feature enabled, but it appears that feature is not recorded in [`rustc_feature::Features`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_feature/struct.Features.html) as well! That's odd 😅, normally you can check it with `cx.tcx.features()`, but well...)

Reply:
Thanks a lot!

This `declared_features` hashset is very insightful, I'd never find it on my own